src/Elewant/Herding/Model/Commands/RenameHerd.php

```diff
@@ -26,4 +30,11 @@ public function newHerdName(): string
     {
         return $this->payload['newHerdName'];
     }
+
+    private function protect() {
+        $newHerdName = $this->payload['newHerdName'];
+
+        Assert::lengthBetween($newHerdName,1, 50);
+        Assert::regex($newHerdName, '/^[^\s]+.*[^\s]+$/');
```

Comment:
Didn't we decide to allow only a subset of characters?

Unicode character groups: Letter, Number, Punctuation.
And emoji, though emoji can't be validated properly with a regular expression.

Reply:
Emoji can be validated with regex, but you have to hardcode the list of code points (and remember to add new ones as they are introduced).  https://github.com/urakozz/php-emoji-regex